src/components/Card.js

```diff
@@ -1,75 +1,94 @@
-    this._link = cardData.link;
+class Card {
+  constructor(
+    { name, link, likes, owner, _id },
+    cardSelector,
+    handleImageClick,
+    handleDeleteCardPopup,
+    handleLikeClick,
+    userID
+  ) {
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._imageName = name;
+    this._imageLink = link;
+    this._handleDeleteCardPopup = handleDeleteCardPopup;
+    this._cardOwnerID = owner._id;
+    this._cardId = _id;
+    this._userID = userID;
+    this._handleLikeClick = handleLikeClick;
+    this._likes = likes;
+  }
+
+  _setEventListeners() {
+    this._likeButton.addEventListener("click", () =>
+      this._handleLikeClick(this)
+    );
+    this._cardDeleteButton.addEventListener("click", () =>
+      this._handleDeleteCardPopup(this)
+    );
+
+    this._cardImage = this._cardElement.querySelector(".card__image");
+    this._cardImage.addEventListener("click", () =>
+      this._handleImageClick({ link: this._imageLink, name: this._imageName })
+    );
+    this._fillCardTemplate();
   }
 
   _getTemplate() {
     return document
       .querySelector(this._cardSelector)
-      .content.querySelector(".card__list-item")
+      .content.querySelector("#cards-template-list")
       .cloneNode(true);
   }
 
-  _fillCardData() {
-    this._cardImage.alt = this._name;
-    this._cardImage.src = this._link;
-    this._cardTitle.textContent = this._name;
-  }
-
-  _setEventListeners() {
-    this._likeButton.addEventListener("click", () => this._handleLikeIcon());
-    this._deleteButton.addEventListener("click", () =>
-      this._handleDeleteCard()
-    );
-
+  _fillCardTemplate() {
     this._cardImage = this._cardElement.querySelector(".card__image");
-    this._cardImage.addEventListener("click", () =>
-      this._handleImageClick({ link: this._link, name: this._name })
-    );
-    // this._cardImage.addEventListener("click", () =>
-    //   this._handlePreviewPicture()
-    // );
+    this._cardImage.alt = this._profName;
+    this._cardImage.src = this._imageLink;
+    this._cardTitle.textContent = this._imageName;
+    this._cardElement.id = this._userID;
+    this._renderLikes();
   }
 
-  _handleLikeIcon() {
-    this._likeButton.classList.toggle("card__like-button_active");
+  _removeDeleteButton() {
+    this._cardDeleteButton.remove();
   }
 
-  _handleDeleteCard() {
+  removeDeleteCard() {
     this._cardElement.remove();
     this._cardElement = null;
   }
 
-  // _handlePreviewPicture() {
-  //   openModal(previewImageModal);
-  //   modalImage.src = this._cardImage.src;
-  //   modalImage.alt = this._cardImage.alt;
-  //   modalCaption.textContent = this._cardTitle.textContent;
-  // }
+  isLiked() {
+    return this._likes.some((like) => like._id === this._userID);
+  }
+  setLikes(numOfLikes) {
+    this._likes = numOfLikes;
+    this._renderLikes();
+  }
+
+  _renderLikes() {
+    this._cardElement.querySelector(".card__like-counter").textContent =
```

Comment:
`this._cardElement.querySelector(".card__like-counter")`

Please, find it once in line 87

**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.